cpp/include/cugraph_c/sampling_algorithms.h

```diff
@@ -187,8 +187,75 @@ typedef struct {
   int32_t align_;
 } cugraph_sample_result_t;
 
+/**
+ * @brief     Opaque sampling options type
+ */
+typedef struct {
+  int32_t align_;
+} cugraph_sampling_options_t;
+
+/**
+ * @brief   Create sampling options object
+ *
+ * All sampling options set to FALSE
+ *
+ * @param [out] options Opaque pointer to the sampling options
+ * @param [out] error   Pointer to an error object storing details of any error.  Will
+ *                      be populated if error code is not CUGRAPH_SUCCESS
+ */
+cugraph_error_code_t cugraph_sampling_options_create(cugraph_sampling_options_t** options,
+                                                     cugraph_error_t** error);
+
+/**
+ * @brief   Set flag to sample with_replacement
+ *
+ * @param options - opaque pointer to the sampling options
+ * @param value - Boolean value to assign to the option
+ */
+void cugraph_sampling_set_with_replacement(cugraph_sampling_options_t* options, bool_t value);
```

Comment:
This is a general question but couldn't we have a single function like below instead of having multiple function calls?
```
cugraph_sampling_set_options(
       cugraph_sampling_options_t* options,
       bool_t with_replacement,
       bool_t with_return_hops,
       bool_t with_unique_sources,
       bool_t with_carry_over_sources,
       bool_t with_dedupe_sources)
```

Reply:
@ChuckHastings

Reply:
First comment... these changes are actually from PR #3696, so it might be a better question there.  But I'll address it here.

We could.  I considered that.  Interestingly, I ended up at this because some of the parameters don't make sense to set together.  In #3696 we are combining these into an enum, so a single call might make more sense.  I will consider that while refactoring that PR.